### PR TITLE
ENG-91: Adding owasp set to jenkins pipeline

### PR DESCRIPTION
**Pull request by jknight-liatrio**

### CODE REVIEW

**Comment by jknight-liatrio** [comment]
I'm not sure where the best place to add this step is, so I've added in 2 steps similar to the selenium tests.

**Comment by jknight-liatrio** [comment]
I've changed the ZAP config to IGNORE existing warnings as pipelines don't pass without it.

File: Jenkinsfile [hunk 1]
```diff
@@ -100,6 +100,23 @@ pipeline {
...
+        stage('Run OWASP ZAP security framework') {
+            agent {
+                docker {
+                    image 'owasp/zap2docker-stable'
+                    args '-u root:sudo --net demo -v ${WORKSPACE}:/zap/wrk:rw'
+                    reuseNode true
+                }
+            }
+            environment {
+                CONTAINER_HTTP_URL = "http://${APP_NAME}:8080"
```
Comment (ericchapman80) [inline Jenkinsfile]: @RobertKelly @jknight-liatrio - Should we be hard coding the port?  Shouldn't that be a parameter that we pass in?

Comment (jknight-liatrio) [reply Jenkinsfile]: This is specific for the app running in the docker container for testing. I modeled it after the selenium test step.

File: Jenkinsfile [hunk 2]
```diff
@@ -100,6 +100,23 @@ pipeline {
...
+            }
+            steps {
+                script { STAGE = env.STAGE_NAME }
+                sh "/zap/zap-baseline.py -d -a -j -t http://${DEV_IP}/${APP_NAME}/ -m 1 -c owasp_zap.conf -r testreport.html -w testreport.md -x testreport.xml -n owasp_zap.context"
```
Comment (ericchapman80) [inline Jenkinsfile]: @jknight-liatrio - Can we specify the name of the report to be something like OWASPtestreport.html or zaptestreport.html.  Thoughts?

File: Jenkinsfile [hunk 3]
```diff
@@ -215,6 +232,23 @@ pipeline {
                 slackSend channel: env.SLACK_ROOM, color: 'good', message: "Application deployed to http://${DEV_IP}/${DEMO_APP_PATH} - waiting on Smoke Test"
             }
         }
+        stage('OWASP ZAP security framework') {
```
[the remark this answers is not in the document]
Comment (jknight-liatrio) [reply Jenkinsfile]: There were 2 stages, one ran against the docker container and the other ran against the deployed app. I've opted to just run one right before the the selenium smoke tests.

File: Jenkinsfile [hunk 4]
```diff
@@ -215,6 +215,23 @@ pipeline {
...
+            agent {
+                docker {
+                    image 'owasp/zap2docker-stable'
+                    args '-u root:sudo --net demo -v ${WORKSPACE}:/zap/wrk:rw'
```
Comment (stein321) [inline Jenkinsfile]: what's the default working directory for zap? sounds like it's not /zap/wrk ?

**Pull request merged**